packages/transformers/js/src/dependency_collector.rs

```diff
@@ -373,10 +371,31 @@ impl<'a> Fold for DependencyCollector<'a> {
 
       // Track the returned require call to be replaced with a promise chain.
       self.require_node = Some(call.clone());
-      return call;
+      call
+    } else if kind == DependencyKind::Require {
+      // Don't continue traversing so that the `require` isn't replaced with undefined
+      node
+    } else {
+      node.fold_children_with(self)
     }
+  }
 
-    return node.fold_children_with(self);
+  fn fold_unary_expr(&mut self, node: ast::UnaryExpr) -> ast::UnaryExpr {
```

Comment:
Shouldn't `fold_expr` already take care of this? It should run before `fold_unary_expr` I think.

Reply:
`UnaryExpr` is an enum variant of `Expr`, so `Expr` is visited before the `UnaryExpr`.

Do you mean I should move the logic in `fold_unary_expr` into `fold_expr`?

Reply:
I mean, we already handle `UnaryExpr` inside the `fold_expr` visitor, and return without visiting children, so I'm not sure why this additional visitor is needed.

Reply:
> I mean, we already handle UnaryExpr inside the fold_expr visitor

I think you are looking at hoist.rs? All of my current changes are in dependency_collector.rs now

Reply:
Oh right 🙈